File: Class02_01_19520007_VoKhanhAn_Homework01/01_19520007_VoKhanhAn_Buoi01_Bai02.py
# STT 01
# Họ và tên: Võ Khánh An
# Buổi 01 - Bài 02
# Ngôn ngữ: Python 

import random
import math

def init():
    print ('NHAN 1 - CHAY CAU 1')
    print ('NHAN 2 - CHAY CAU 2')
    print ('NHAN 3 - CHAY CAU 3')
    print ('NHAN 4 - CHAY CAU 4')
    print ('NHAN 5 - THOAT KHOI CHUONG TRINH')
    print ('----------------------------------')

def cau1():
    global n
    global a

    for i in range (n):
        print ('a [',i, '] = ', end = ' ')
        a.append(int(input()))

def cau2():
    global n
    global a

    n = random.randint(10, 50)
    for i in range (n):
        a.append(100 + random.randint(100, 120))

def cau3():
    global n
    global a
    print ('n =', n)
    for i in range (0, n):
        print ('a [', i ,'] =', a[i])

def cau4():
    global a
    global n
    a.sort()
    m = 0
    val = []
    cnt = []
    val.append(a[0])
    cnt.append(1)
    for i in range (n-1):
        if a[i] == a[i + 1]:
            cnt[m] = cnt[m] + 1
        else:
            m = m + 1
            val.append(a[i + 1])
            cnt.append(1)
    for i in range (m + 1):
        print ('GIA TRI =', val[i])
        print ('SO LUONG =', cnt[i])
        print ('\n')



a = []

while True:
    init()
    res = int(input())
    if res == 1:
        print ('n = ', end =' ')
        n = int(input())
        cau1()
        print('\n')
    elif res == 2:
        n = 0
        cau2()
        print('\n')
    elif res == 3:
        cau3()
        print('\n')
    elif res == 4:
        cau4()
        print ('\n')
    elif res == 5:
        print ('EXIT')
        exit (0)
    else:
            print ('Ban da nhap sai, vui long nhap lai: ')

File: Class02_01_19520007_VoKhanhAn_Homework01/01_19520007_VoKhanhAn_Buoi01_Bai02.cpp
// STT: 01
// Họ và tên: Võ Khánh An
// Buổi 01 - Bài 02
// Ngôn ngữ: C++

#include <iostream>
#include <stdlib.h>
#include <time.h>
#define MAXN 105

using namespace std;

struct str
{
    long cnt, val;
};


void cau1(long a[], long &n)
{
    cout << "n = ";
    cin >> n;
    for (long i = 0; i < n; i ++)
    {
        cout << "a[" << i << "] = ";
        cin >> a[i];
    }
        
}

void cau2(long a[], long &n)
{
    srand(time(0));
    n = 10 + rand() % 40;
    for (long i = 0; i < n; i++)
    {
        a[i] = 100 + rand() % 20;
    }
}

void cau3(long a[], long n)
{
    cout << "n = " << n << endl;
    for (long i = 0; i < n; i++)
        cout << "a[" << i << "] = " << a[i] << endl;
}

void qsort(long a[], long n, long l, long r)
{
    long x = a[(l + r) / 2], i = l, j = r;
    do {
        while (a[i] < x) i++;
        while (a[j] > x) j--;

        if (i <= j)
        {
            long tm = a[i];
            a[i] = a[j];
            a[j] = tm;
            i++; j--;
        }
    } while (i <= j);

    if (l < j) qsort (a, n, l, j);
    if (i < r) qsort (a, n, i, r);
}

void cau4(long a[], long n)
{
    qsort (a, n, 0 ,n - 1);

    long m = 0; str b[MAXN];
    
    b[0].val = a[0];
    b[0].cnt = 1;

    for (long i = 0; i < n - 1; i++)
        if (a[i] == a[i + 1])
        {
            b[m].cnt++;
        }
        else {
            m++;
            b[m].val = a[i + 1];
            b[m].cnt = 1;
        }
    for (long i = 0; i <= m; i++) {
        cout << "GIA TRI = " << b[i].val << endl;
        cout << "SO LUONG = " << b[i].cnt << endl;
        cout << endl;
    }
}

void init()
{
    cout << "NHAN 1 - CHAY CAU 1" << endl;
    cout << "NHAN 2 - CHAY CAU 2" << endl;
    cout << "NHAN 3 - CHAY CAU 3" << endl;
    cout << "NHAN 4 - CHAY CAU 4" << endl;
    cout << "NHAN 5 - THOAT KHOI CHUONG TRINH" << endl;
    cout << "-------------------" << endl;
    cout << endl;
}

int main()
{
    long n, a[MAXN];
    int res;
    while (true)
    {
        init ();
        cin >> res;
        switch (res)
        {
            case 1:
                cau1(a, n);
                cout << endl;
                break;
            case 2:
                cau2(a, n);
                cout << endl;
                break;
            case 3:
                cau3(a, n);
                cout << endl;
                break;
            case 4:
                cau4(a, n);
                cout << endl;
                break;
            case 5:
                cout <<"EXIT" << endl;
                return 0;
            default:
                cout << "Ban da nhap sai, vui long nhap lai: " << endl;
                cout << endl;
        }
    }
}